quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -443,6 +461,45 @@ impl IndexingService {
         self.split_stores.insert(key, split_store.downgrade());
         Ok(split_store)
     }
+
+    async fn get_or_init_merge_pipeline(
+        &mut self,
+        ctx: &ActorContext<Self>,
+        pipeline_id: &IndexingPipelineId,
+        doc_mapper: Arc<dyn DocMapper>,
+        indexing_directory: IndexingDirectory,
+        split_store: IndexingSplitStore,
+        merge_policy: Arc<dyn MergePolicy>,
+    ) -> anyhow::Result<Mailbox<MergePlanner>> {
+        let key = (pipeline_id.index_id.clone(), pipeline_id.source_id.clone());
+        if let Some((merge_planner_mailbox, _)) = self.merge_pipeline_handles.get(&key) {
```

Comment:
This could be the mailbox of a dead merge pipeline if we are post merge pipeline crash but pre supervision loop. A merge pipeline crash will also crash all the indexing pipelines. I wonder if we should recycle the merge planner mailbox between merge pipelines respawn. @fmassot, what is the status of recycling actors' mailboxes/inboxes?

Reply:
- An idea I had was to probe the `merge_planner_mailbox` with a dummy message before progressing with it? Otherwise, it should be noticed on the first split produced and respawn. 
- I think having the indexing pipelines crash when their common merge pipeline crash is fair. The indexing pipelines will respawn and the first to succeed will create a valid merge pipeline others can use. 
It would be awesome :100:  to recycle mailboxes thought: that is allowing the  `merge_planner_mailbox` to accept messages even if its consumer end is dead and being able to reconnect to a new consumer (merge pipeline)

Reply:
I'd say it's fair-ish. Crashing pipelines wastes resources, i.e., money :) We should be able to sustain the unavailabilty of the merge pipeline and keep publishing splits.

Reply:
@guilload, I am now checking if the merge_pipeline is healthy and cleaning before proceeding.